diff -ruNp lxqt-config-orig/lxqt-config-input/CMakeLists.txt lxqt-config/lxqt-config-input/CMakeLists.txt
--- lxqt-config-orig/lxqt-config-input/CMakeLists.txt	2025-02-22 11:12:36.000000000 +0330
+++ lxqt-config/lxqt-config-input/CMakeLists.txt	2025-02-23 18:59:57.296953387 +0330
@@ -35,6 +35,7 @@ set(lxqt-config-input_HDRS
     mouseconfig.h
     keyboardlayoutconfig.h
     selectkeyboardlayoutdialog.h
+    ../lxqt-config-appearance/configothertoolkits.h
 )
 
 set(lxqt-config-input_SRCS
@@ -43,6 +44,7 @@ set(lxqt-config-input_SRCS
     mouseconfig.cpp
     keyboardlayoutconfig.cpp
     selectkeyboardlayoutdialog.cpp
+    ../lxqt-config-appearance/configothertoolkits.cpp
 )
 
 set(lxqt-config-input_UIS
diff -ruNp lxqt-config-orig/lxqt-config-input/lxqt-config-input.cpp lxqt-config/lxqt-config-input/lxqt-config-input.cpp
--- lxqt-config-orig/lxqt-config-input/lxqt-config-input.cpp	2025-02-22 11:19:00.000000000 +0330
+++ lxqt-config/lxqt-config-input/lxqt-config-input.cpp	2025-02-23 18:53:22.403605742 +0330
@@ -25,6 +25,7 @@
 #include "mouseconfig.h"
 #include "keyboardconfig.h"
 #include "../liblxqt-config-cursor/selectwnd.h"
+#include "../lxqt-config-appearance/configothertoolkits.h"
 #include "keyboardlayoutconfig.h"
 
 #ifdef WITH_TOUCHPAD
@@ -79,6 +80,14 @@ int main(int argc, char** argv) {
 
     LXQt::Settings qtSettings(QStringLiteral("lxqt"));
 
+    LXQt::Settings configAppearanceSettings(QStringLiteral("lxqt-config-appearance"));
+    ConfigOtherToolKits *configOtherToolKits = nullptr;
+    if (configAppearanceSettings.value(QStringLiteral("ControlGTKThemeEnabled")).toBool())
+    {
+        configOtherToolKits = new ConfigOtherToolKits(&qtSettings, &configAppearanceSettings, &dlg);
+        configOtherToolKits->startXsettingsd();
+    }
+
     /*** Mouse Config ***/
     MouseConfig* mouseConfig = new MouseConfig(&settings, &qtSettings, &dlg);
     dlg.addPage(mouseConfig, QObject::tr("Mouse"), QStringLiteral("input-mouse"));
@@ -135,6 +144,17 @@ int main(int argc, char** argv) {
         {
             mouseConfig->applyConfig();
             cursorConfig->applyCusorTheme();
+            if (configOtherToolKits)
+            {
+                // GTK3
+                configOtherToolKits->setGTKConfig(QStringLiteral("3.0"));
+                if(QGuiApplication::platformName() == QStringLiteral("wayland"))
+                    configOtherToolKits->setGsettingsConfig(QStringLiteral("3.0"));
+                // GTK2
+                configOtherToolKits->setGTKConfig(QStringLiteral("2.0"));
+                // Update xsettingsd
+                configOtherToolKits->setXSettingsConfig();
+            }
             keyboardConfig->applyConfig();
             if (keyboardLayoutConfig)
                 keyboardLayoutConfig->applyConfig();
